app/src/tunnel/kotlin/tunnel/FilterSerializer.kt

```diff
@@ -9,16 +12,17 @@ class FilterSerializer {
         val filters = repo.asSequence().batch(9).map { entry ->
             entry[0].toInt() to try {
                 val id = entry[1]
+                val wildcard = entry[2] == "wildcard"
```

Comment:
So if I added another element so list looked like the below


3
b_t2custom
**wildcard**
blacklist
active
https://https://gist.github.com/Thomas499
link
https://gist.githubusercontent.com/Thomas499/cb5dc1c2e515d556518241ba1c194cfd/raw

What would you call the normal list? The ones that are't "wildcard"? would you call it "normal"?

Reply:
"normal" or leaf it empty. it's your choice. You could also just change the way the entry is parsed. So that it can be "whitelist", "blacklist" or "wildcardblacklist" or something like that. This would be better because 3.x versions would just parse it as a normal blacklist which has no valid hosts in it.

Reply:
>  So that it can be "whitelist", "blacklist" or "wildcardblacklist" or something like that.

Wouldn't that still cause the issue of:

> Using the same entry to distinguish between wildcards and whitelists this way is problematic. 

? 

It might be possible to do something like
var listtype=0
if (entry[2] == "wildcardblacklist")
 listype=1
else if (entry[2] == "wildcardwhitelist")
listype=2
else if (entry[2]== "whitelist")
listtype=3
else
listtype=5

that would be easy enough to work with. What are your thoughts?

Reply:
You can do it that way. Yes. However, I would still use two separate variables for white/blacklist and wildcard/no wildcard.

(Please use the `code`-format if you are adding source code drafts to your comments. Even if they're just pseudo code. It's much more readable.)

Reply:
![image](https://user-images.githubusercontent.com/38442058/59130900-964dff00-8925-11e9-85f4-49f8ee319c60.png)

If I added a new line, for whitelist/no whitelist, then wouldn't it mess up older versions of the app if the person were to clear data, resulting in the app looking for this list again and finding a line that previously wasn't supposed to be there?

Reply:
yes. That's why you should use the same line. That's what I meant when i sayed:

> This would be better because 3.x versions would just parse wildcard-files as a normal blacklist which has no valid hosts in it.

The other possibility would be a new backend version under a different URL but that would be the less favorable version.